src/libraries/System.Runtime/tests/System/GenericMathTests.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Tests
+{
+    public static class GenericMath
+    {
+        public static TResult Average<TSelf, TResult>(IEnumerable<TSelf> values)
+            where TSelf : INumber<TSelf>
+            where TResult : INumber<TResult>
+        {
+            TResult sum = Sum<TSelf, TResult>(values);
+            return TResult.Create(sum) / TResult.Create(values.Count());
```

Comment:
Just adding a note that once we start consuming the DetectPreviewFeaturesAnalyzer, we'd need to opt into preview features in this csproj.